README.markdown

```diff
@@ -47,8 +44,75 @@ architectural decisions enable this:
    The only place state is held is in the Ceph cluster (which, by
    definition, is consistent and redundant).
 
-Implementation
---------------
+
+# Deployment
+
+## System dependencies
+
+ - [librados](http://ceph.com)
+ - [zeromq 4+](http://zeromq.org/)
+ - [GHC 7.8.3+](https://www.haskell.org/ghc/)
+ - [cabal](https://www.haskell.org/cabal/)
+
+## Common packages
+
+ - [vaultaire-common](https://github.com/anchor/vaultaire-common)
+ - [marquise](https://github.com/anchor/marquise)
+
+## Build
+
+```
+cabal sandbox init
+cabal sandbox add-source ../vaultaire-common
+cabal sandbox add-source ../marquise
+cabal install
+```
+
+You'll need to add the sandbox's `bin` directory to your `$PATH`, or
+else replace the `vault` command with a fully-qualified path.
+
+## Running
+
+You need to run four components: the `broker`, plus one or more of each
+of the `writer`, `reader` and `contents` daemons. These can be on the
+same or different hosts, but they must all point to the same broker (if
+you want to make the broker highly-available, use a TCP load-balancer).
+
+The defaults assume your Rados pool is called `vaultaire`, and you're
+connecting with the username `vaultaire`; if this isn't the case,
+override with `-p` and `-u` respectively. In either case, you'll need to
+have the environment variable `CEPH_KEYRING` pointing to the correct
+place.
+
+```
+vault -d broker # Start broker in debug mode
+vault -d writer -b localhost # Start writer in debug mode
+vault -d reader -b localhost # Start reader in debug mode
+vault -d contents -b localhost # Start contents daemon in debug mode
+```
+
+# Ecosystem
+
+Vaultaire on its own only provides a binary interface over ZeroMQ. To
```

Comment:
@fractalcat This is excellent writing. When you anticipate a [new] users' questions and answer them preemtively it gives the impression that they're in good hands. That's vital for adoption, but even where we aren't per-se trying to convince people to run their own Vaultaires we do want people to feel we know what we're about. This is exactly the sort of things that encourages that.